### ISSUE TITLE
Widget isn't visible in Release mode build

### ISSUE THREAD

**Issue by nammaapp** [ikicodedev/calendar_timeline#35]
This widget isn't visible when I tried to run in release mode, all I get is below message in output window.

I/flutter (19247): Unsupported operation: Infinity or NaN toInt
I/flutter (19247): #0      State.widget (package:flutter/src/widgets/framework.dart)
I/flutter (19247): #1      _PositionedListState._schedulePositionNotificationUpdate.<anonymous closure> (package:scrollable_positioned_list/src/positioned_list.dart:348)
I/flutter (19247): #2      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1145)
I/flutter (19247): #3      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1090)
I/flutter (19247): #4      SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:863)
I/flutter (19247): #12     TickerFuture.whenCompleteOrCancel.thunk (package:flutter/src/scheduler/ticker.dart:407)
I/flutter (19247): (elided 10 frames from class _RawReceivePortImpl, class _Timer, and dart:async)
I/flutter (19247): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'
I/flutter (19247): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'

This seems to be the case even with Example code at https://pub.dev/packages/calendar_timeline/example.

Debug mode works fine without any issues. Any help would be highly appreciated. thanks.

**Comment by ikicodedev** [comment]
Hi @nammaapp, thanks for reporting the bug. I'm trying to reproduce the error on a real device in release mode and I can't. Can you give me more information about the error? Device, OS version, etc.

**Comment by truly6ve** [comment]
Hi,

I also have the same issue in the release version of the website after launch. Please check the attached images showing the issue. In the first one it worked all fine while debugging and in the second one after release, it didn't display anything. I have tried it on Chrome and Safari both have issues after release.

Regards,

<img width="991" alt="Screenshot 2022-01-26 at 13 12 24" src="https://user-images.githubusercontent.com/72092249/151169379-a0bc6daf-1a1d-4c6c-874b-1d3808311304.png">
<img width="991" alt="Screenshot 2022-01-26 at 13 11 26" src="https://user-images.githubusercontent.com/72092249/151169385-db0beb55-4f7b-4860-b2a0-e9159e3650ea.png">

**Comment by ikicodedev** [comment]
Hi @truly6ve, What error does it show in the console? What flutter version do you use?

**Comment by nammaapp** [comment]
> Hi,
> 
> I also have the same issue in the release version of the website after launch. Please check the attached images showing the issue. In the first one it worked all fine while debugging and in the second one after release, it didn't display anything. I have tried it on Chrome and Safari both have issues after release.
> 
> Regards,
> 
> <img alt="Screenshot 2022-01-26 at 13 12 24" width="991" src="https://user-images.githubusercontent.com/72092249/151169379-a0bc6daf-1a1d-4c6c-874b-1d3808311304.png">
> 
> <img alt="Screenshot 2022-01-26 at 13 11 26" width="991" src="https://user-images.githubusercontent.com/72092249/151169385-db0beb55-4f7b-4860-b2a0-e9159e3650ea.png">

Hi @truly6ve, just call setState(() {}) once after everything loaded, that should fix the issue as a temporary solution. In my case, I moved 'data loading' part into a FutureBuilder and which somehow repaints the calendartimeline again in release mode.

**Comment by nammaapp** [comment]
> Hi @nammaapp, thanks for reporting the bug. I'm trying to reproduce the error on a real device in release mode and I can't. Can you give me more information about the error? Device, OS version, etc.

Hi ikicodedev, I receive this error in my samsung S8 plus (SM-G955FD) phone, but worked fine on other mobile devices I tried. If you look the stack trace I previously mentioned, the exact error occurs at package:scrollable_positioned_list/src/positioned_list.dart at line 348,  positions.add(ItemPosition(... at else {} part. 

I tried checking by adding simple print() statements and found that "itemOffset" variable somehow becomes NaN in release mode.

final itemOffset = box.localToGlobal(Offset.zero, ancestor: viewport).dx;
print(itemOffset);

I/flutter (19507): NaN

I'm not a flutter expert but feeling like this error might cause painting to stop. Somehow calling a blank setstate() repaints the control properly. I added a Button and onclicking it, just called setstate() function which made the calendartimeline to reappear in release mode.
